lib/src/default_index/rev_walk.rs

```diff
@@ -248,6 +249,17 @@ pub(super) trait RevWalkIndex {
     type AdjacentPositions: IntoIterator<Item = Self::Position>;
 
     fn adjacent_positions(&self, pos: Self::Position) -> Self::AdjacentPositions;
+
+    fn adjacent_positions_in_range(
```

Comment:
nit: can you extract this to free function? I don't think trait method is needed.

BTW, I think it's simpler to use slice than `.skip()` and `.take()`.